File: Domain/Models/Vehiculos/Procedencia.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.Models.Vehiculos
{
    public class Procedencia
    {
        [Key]
        public int? ProcedenciaId { get; set; }

        [Column(TypeName = "nvarchar(20)")]
        public string Descripcion { get; set; }
    }
}

File: CarAgency/ClientApp/src/domain/models/vehiculos/Procedencia.ts
import  IProcedencia from "domain/interfaces/vehiculos/IProcedencia";

export default class Procedencia implements IProcedencia {
    constructor(){
        this.ProcedenciaId = 0;
        this.Descripcion = "";
    }
    
    ProcedenciaId: number;
    Descripcion: string;

}